intake_esm/core.py

```diff
@@ -278,6 +292,12 @@ def _get_subset(self, **query):
         query_results = self.df.loc[condition]
         return query_results.reset_index(drop=True)
 
+    def _clear_old_catalog(self, catalog_data):
+        """ Remove any old references to the catalog."""
+        for key in {'catalog_dict', 'catalog_file'}:
+            _ = catalog_data.pop(key, None)
+        return catalog_data
```

Comment:
:+1: 

I have one suggestion:  this function doesn't need to be an instance method (i.e have self as a parameter) since we are not really using any extra information from the class. I propose that we remove the `self` parameter and make this function a  private/nested function in the `serialize()` method:

```python
def serialize(self, name, directory=None, catalog_type='dict'):

  def _clear_old_catalog(catalog_data):
        """ Remove any old references to the catalog."""
        for key in {'catalog_dict', 'catalog_file'}:
            _ = catalog_data.pop(key, None)
        return catalog_data

  collection_data = _clear_old_catalog(collection_data)
...
```

What do you think?

Reply:
@andersy005 I think it's a good idea.  Do private functions usually go at the top of the parent function?

Reply:
> Do private functions usually go at the top of the parent function?

It's not required. They can be put anywhere. The only requirement is that by the time you call the function, it must be defined/declared.